lib/named-locks.js

```diff
@@ -45,7 +45,27 @@ const sql = sqlLoader.loadSqlEquiv(__filename);
  * form `{client, done}`, where `client` and `done` are sqldb
  * transaction objects. These are simply the objects we need to end
  * the transaction, which will release the lock.
+ *
+ * Importantly, we use a separate pool of database connections for acquiring
+ * and holding locks. This ensures that we don't end up with a deadlock.
+ * For instance, if we have a pool of 10 clients and there are 10 locks held at
+ * once, if any code inside of a lock tries to acquire another database client,
+ * we'd deadlock if we weren't separating the two pools of connections.
+ */
```

Comment:
Should we add a warning somewhere that we must not nest locks? Or do we think it's ok to nest them, given the AsyncLocalStorage code?

Reply:
Until we can remove everything in favor of `doWithLock()`, we can't rely on ALS to work correctly at all here, unfortunately. I'll add a comment.

Reply:
Added a note in d7bb094.

Reply:
Looks good.